src/style-spec/read_style.js

```diff
@@ -0,0 +1,15 @@
+import ParsingError from './error/parsing_error';
+import jsonlint from '@mapbox/jsonlint-lines-primitives';
+
+export default function readStyle(style) {
+    let s = style;
+    if (s instanceof String || typeof s === 'string' || s instanceof Buffer) {
+        try {
+            s = jsonlint.parse(s.toString());
+        } catch (e) {
+            throw new ParsingError(e);
+        }
+    }
+
+    return s;
```

Comment:
What's the case for returning the original input without attempting to parse? Could we remove the `if` statement and simplify this function to:

```js
export default function readStyle(style) {
  try {
    return jsonlint.parse(style.toString());
  } catch (e) {
    throw new ParsingError(e);
  }
}

Reply:
I imagine it's just a performance optimization. It was in the code before this PR so I am inclined to leave it since validate _is_ run at runtime by GL JS.